crypto/zkp/primitives/commitments/damgard-fujisaki_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017 XLAB d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package commitmentzkp
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/xlab-si/emmy/crypto/commitments"
+	"github.com/xlab-si/emmy/crypto/common"
+	"math/big"
+	"testing"
+)
+
+// TestProveDFCommitmentOpening demonstrates how to prove that you can open DamgardFujisaki commitment.
+func TestProveDFCommitmentOpening(t *testing.T) {
+	receiver, err := commitments.NewDamgardFujisakiReceiver(1024, 80)
+	if err != nil {
+		t.Errorf("Error in NewDamgardFujisakiReceiver: %v", err)
+	}
+
+	committer := commitments.NewDamgardFujisakiCommitter(receiver.QRSpecialRSA.N,
+		receiver.H, receiver.G, receiver.K)
+
+	x := common.GetRandomInt(committer.T)
+	c, err := committer.GetCommitMsg(x)
+	if err != nil {
+		t.Errorf("Error in computing commit msg: %v", err)
+	}
+	receiver.SetCommitment(c)
+
+	challengeSpaceSize := 80
+	prover := NewDFCommitmentOpeningProver(committer, challengeSpaceSize)
+	verifier := NewDFCommitmentOpeningVerifier(receiver, challengeSpaceSize)
+
+	proofRandomData := prover.GetProofRandomData()
+	verifier.SetProofRandomData(proofRandomData)
+
+	challenge := verifier.GetChallenge()
+	s1, s2 := prover.GetProofData(challenge)
+	proved := verifier.Verify(s1, s2)
+
+	assert.Equal(t, true, proved, "DamgardFujisaki opening proof failed.")
+}
+
+// TestProveDFCommitmentMultiplication demonstrates how to prove that for given commitments
+// c1 = g^x1 * h^r1, c2 = g^x2 * h^r2, c3 = g^x3 * h^r3, it holds x3 = x1 * x2
+func TestProveDFCommitmentMultiplication(t *testing.T) {
```

Comment:
I suggest renaming this test to something that includes `DamgardFujisaki`. For instance, `TestProveDamgardFujisakiCommitmentMultiplication`. It is indeed longer but produces a clearer output when the tests are actually run.